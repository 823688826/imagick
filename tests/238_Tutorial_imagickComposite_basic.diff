001+ Fatal error: Uncaught exception 'ImagickException' with message 'Can not process empty Imagick object' in /home/github/mkoppanen-imagick/tests/238_Tutorial_imagickComposite_basic.php:11
001- Ok
002+ Stack trace:
003+ #0 /home/github/mkoppanen-imagick/tests/238_Tutorial_imagickComposite_basic.php(11): Imagick->getimagewidth()
004+ #1 /home/github/mkoppanen-imagick/tests/238_Tutorial_imagickComposite_basic.php(40): imagickComposite()
005+ #2 {main}
006+   thrown in /home/github/mkoppanen-imagick/tests/238_Tutorial_imagickComposite_basic.php on line 11