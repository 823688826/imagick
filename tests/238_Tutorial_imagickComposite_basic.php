<?php


function imagickComposite() {
    //Load the images
    $left = new \Imagick(realpath('images/im/holocaust_tn.gif'));
    $right = new \Imagick(realpath('images/im/spiral_stairs_tn.gif'));
    $gradient = new \Imagick(realpath('images/im/overlap_mask.png'));

    //The right bit will be offset by a certain amount - avoid recalculating.
    $offsetX = $gradient->getImageWidth() - $right->getImageWidth();


    //Fade out the left part - need to negate the mask to
    //make math correct
    $gradient2 = clone $gradient;
    $gradient2->negateimage(false);
    $left->compositeimage($gradient2, \Imagick::COMPOSITE_COPYOPACITY, 0, 0);

    //Fade out the right part
    $right->compositeimage($gradient, \Imagick::COMPOSITE_COPYOPACITY, -$offsetX, 0);

    //Create a new canvas to render everything in to.
    $canvas = new \Imagick();
    $canvas->newImage($gradient->getImageWidth(), $gradient->getImageHeight(), new \ImagickPixel('black'));

    //Blend left half into final image
    $canvas->compositeimage($left, \Imagick::COMPOSITE_BLEND, 0, 0);

    //Blend Right half into final image
    $canvas->compositeimage($right, \Imagick::COMPOSITE_BLEND, $offsetX, 0);

    //Output the final image
    $canvas->setImageFormat('png');

    $bytes = $canvas->getImageBlob();
    if (strlen($bytes) <= 0) { echo "Failed to generate image.";} 
}

imagickComposite() ;
echo "Ok";
?>
