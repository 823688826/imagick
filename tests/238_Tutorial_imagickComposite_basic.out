Fatal error: Uncaught exception 'ImagickException' with message 'Can not process empty Imagick object' in /home/github/mkoppanen-imagick/tests/238_Tutorial_imagickComposite_basic.php:11
Stack trace:
#0 /home/github/mkoppanen-imagick/tests/238_Tutorial_imagickComposite_basic.php(11): Imagick->getimagewidth()
#1 /home/github/mkoppanen-imagick/tests/238_Tutorial_imagickComposite_basic.php(40): imagickComposite()
#2 {main}
  thrown in /home/github/mkoppanen-imagick/tests/238_Tutorial_imagickComposite_basic.php on line 11